lib/3scale/client.rb

```diff
@@ -285,6 +287,13 @@ def encode_transactions(transactions)
       result
     end
 
+    def usage_query_params(usage)
+      usage.map do |metric, value|
+        escaped_metric = CGI.escape "[usage][#{metric}]"
+        "#{escaped_metric}=#{CGI.escape(value.to_s)}"
+      end.join('&')
```

Comment:
I just extracted the method as it was. But yes, while we are at it I can change it. Your solution looks cleaner.